core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -384,18 +498,23 @@ class Partition(val topicPartition: TopicPartition,
    * from the time when this broker was the leader last time) and setting the new leader and ISR.
    * If the leader replica id does not change, return false to indicate the replica manager.
    */
-  def makeLeader(controllerId: Int, partitionStateInfo: LeaderAndIsrRequest.PartitionState, correlationId: Int): Boolean = {
+  def makeLeader(controllerId: Int,
+                 partitionStateInfo: LeaderAndIsrRequest.PartitionState,
+                 correlationId: Int,
+                 highWatermarkCheckpoint: LazyOffsetCheckpoints): Boolean = {
```

Comment:
Should we add a simple trait for LazyOffsetCheckpoints? I see that it gets used in lots of method calls, but would probably simplify testing it we had an interface to provide `fetch(String, TopicPartition): Option[Long]`